lib/Conversion/TorchToMhlo/Linear.cpp

```diff
@@ -412,33 +412,55 @@ class ConvertAtenLinearOp : public ConvertAtenMatmulBaseOp<AtenOpT> {
 
     auto lhsTy = lhs.getType().cast<RankedTensorType>();
     auto rhsTy = rhs.getType().cast<RankedTensorType>();
-    auto leadingRank = std::max(lhsTy.getRank() - rhsTy.getRank(),
-                                rhsTy.getRank() - lhsTy.getRank());
-
-    const auto &options = ConvertAtenOp<AtenOpT>::getOptions();
-    getBmmBroadcast(rewriter, op, lhs, rhs, leadingRank,
-                    options.dimSizeIndexBits);
-    auto resultRank = std::max(lhsTy.getRank(), rhsTy.getRank());
-    auto nBatchDims = resultRank - 2;
-    auto batchDims = llvm::to_vector<4>(llvm::seq<int64_t>(0, nBatchDims));
-
-    auto lhsResultDim = nBatchDims;
-    auto rhsResultDim = nBatchDims + 1;
-    auto lhsContractingDim = nBatchDims + 1;
-    auto rhsContractingDim = nBatchDims;
-
+    auto lhsRank = lhsTy.getRank();
+    auto rhsRank = rhsTy.getRank();
+    if (rhsRank != 2)
+      return op.emitError("aten.linear weights must have rank 2");
+    auto loc = op->getLoc();
+    Value dotLhs;
+    SmallVector<Value> resultDims;
+    // vector * matrix or matrix * matrix can directly use mhlo.dot_general
+    if (lhsTy.getRank() <= 2) {
+      dotLhs = lhs;
+    } else {
```

Comment:
Nit： how about add some rationale reasons why we choose to do reshape? What is its advantages to do broadcast multiply directly?